### PR TITLE
getAccountHistory: don't wait for uniswap

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
tested on local prod API with sim -- sends push notifs still, shows getAccountHistory fast

### CODE REVIEW

File: packages/daimo-common/src/model.ts [hunk 1]
```diff
@@ -128,6 +128,7 @@ export interface ProposedSwap {
   fromAcc: EAccount;
   receivedAt: number;
   toAmount: number; // in native USDC units
+  expiresAt: number;
```
Comment (dcposch) [inline packages/daimo-common/src/model.ts:131]: maybe cacheUntil to distinguish from deadline

**Review by dcposch — APPROVED** [review]
LGTM after comments

File: apps/daimo-mobile/src/logic/inAppNotifications.ts [hunk 2]
```diff
@@ -99,6 +100,9 @@ export function useInAppNotifications(): InAppNotificationInfo {
 
     // Proposed swaps
     account.proposedSwaps.map((swap) => {
+      if (swap.expiresAt < now()) {
+        resync("Refetching swap data");
```
[the remark this answers is not in the document]
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/logic/inAppNotifications.ts]: done

File: packages/daimo-api/src/network/uniswapClient.ts [hunk 3]
```diff
@@ -77,10 +79,26 @@ export class UniswapClient {
...
     if (swap == null) return;
-    const key = `${addr}-${swap.fromAmount}-${swap.fromCoin.token}-${swap.receivedAt}-${swap.fromAcc.addr}`;
-    this.swapCache.set(key, [swap, now()]);
+    const key = this.swapCacheKey(
+      addr,
+      swap.fromAmount,
+      swap.fromCoin.token,
+      swap.receivedAt,
+      swap.fromAcc.addr
+    );
```
[the remark this answers is not in the document]
Comment (nalinbhardwaj) [reply packages/daimo-api/src/network/uniswapClient.ts:100]: done

File: packages/daimo-api/src/network/uniswapClient.ts [hunk 4]
```diff
@@ -89,45 +107,31 @@ export class UniswapClient {
...
-      fromAcc.addr
-    );
-    if (cachedSwap) {
-      console.log(`[UNISWAP] using cached swap ${cachedSwap.fromCoin.token}`);
-      return cachedSwap;
-    }
+  ) {
+    assert(this.router != null, "Uniswap router not initialized");
 
+    const expiresAt = now() + 1800; // 30 min
```
[the remark this answers is not in the document]
Comment (nalinbhardwaj) [reply packages/daimo-api/src/network/uniswapClient.ts]: done

File: packages/daimo-api/src/network/uniswapClient.ts [hunk 5]
```diff
@@ -89,45 +107,31 @@ export class UniswapClient {
...
     const options: SwapOptionsSwapRouter02 = {
       recipient: addr,
       slippageTolerance: new Percent(50, 10_000), // 50 bips
-      deadline: Math.floor(now() + 600),
+      deadline: expiresAt + 60, // 1 extra minute for buffer
```
[the remark this answers is not in the document]
Comment (nalinbhardwaj) [reply packages/daimo-api/src/network/uniswapClient.ts]: made deadline as above

File: packages/daimo-api/src/network/uniswapClient.ts [hunk 6]
```diff
@@ -159,14 +167,62 @@ export class UniswapClient {
...
+  ): Promise<ProposedSwap | null> {
+    const cachedSwap = this.getCachedSwap(
+      addr,
+      fromAmount,
+      fromCoin.token,
+      receivedAt,
+      fromAcc.addr
+    );
+
+    if (!cachedSwap || cachedSwap.expiresAt < now()) {
```
[the remark this answers is not in the document]
Comment (nalinbhardwaj) [reply packages/daimo-api/src/network/uniswapClient.ts]: done

File: packages/daimo-common/src/model.ts (hunk 1 above)
Comment (nalinbhardwaj) [reply packages/daimo-common/src/model.ts]: done

**Pull request merged**